polar-core/src/parser.rs

```diff
@@ -228,13 +228,18 @@ mod tests {
 
     #[test]
     fn test_parse_new() {
-        let f = r#"
-        a(x) if x = new Foo{a: 1};
-        "#;
+        let f = r#"a(x) if x = new Foo{a: 1};"#;
         let results = parse_rules(0, f).unwrap();
         assert_eq!(results[0].to_polar(), r#"a(x) if x = new Foo{a: 1};"#);
     }
 
+    #[test]
+    fn test_parse_new_boa_constructor() {
+        let f = r#"a(x) if x = new Foo(1, 2);"#;
+        let results = parse_rules(0, f).unwrap();
+        assert_eq!(results[0].to_polar(), r#"a(x) if x = new Foo(1, 2);"#);
+    }
+
```

Comment:
Could we include a test of the error that results from attempting to call with a combination of positional and kwargs? E.g., `Foo(1, bar: 2)`. Not sure how common it would be but if the syntax is similar enough to Ruby people might try it and it should have a decent error.

Reply:
Ugh. These errors are not handled well at all, but it's not specific to constructors. Can we please make this a follow-up task? E.g.,
```
python -m oso
query> f(x:x)
Traceback (most recent call last):
  File "/home/alex/oso/oso/languages/python/polar/polar.py", line 124, in repl
    ffi_query = self.ffi_polar.new_query_from_str(query)
  File "/home/alex/oso/oso/languages/python/polar/ffi.py", line 30, in new_query_from_str
    check_result(lib.polar_new_query(self.ptr, to_c_str(query_str), 0))
  File "/home/alex/oso/oso/languages/python/polar/ffi.py", line 108, in check_result
    raise Error().get()
polar.exceptions.UnrecognizedToken: did not expect to find the token ':' at line 1, column 4

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/alex/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/alex/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/alex/oso/oso/languages/python/oso/__main__.py", line 3, in <module>
    Oso().repl(load=True)
  File "/home/alex/oso/oso/languages/python/polar/polar.py", line 126, in repl
    print("Parse error: ", str(e.value()))
AttributeError: 'UnrecognizedToken' object has no attribute 'value'
```

That said, I think this is the right syntax for us to use for mixed positional/keyword args. I think it's more consistent with the rest of the language than a Python-style `f(a=b)`.

Reply:
Sure, can take this as a follow-up. That error is bad, but that's not actually the syntax I meant to check. I meant something like `query> x = new Foo(id: 1)`.

Reply:
This is what I get with that, which actually isn't too bad: 

```
polar = <polar.polar.Polar object at 0x10b5da610>, qeval = <function qeval.<locals>._qeval at 0x10b6c0d40>

    def test_boa_constructor(polar, qeval):
        class Foo:
            def __init__(self, id=0):
                self.id = 1
    
        polar.register_class(Foo)
>       polar.load_str("f(x) if x = new Foo(id: 1);")

tests/test_polar.py:663: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
polar/polar.py:66: in load_str
    self.ffi_polar.load_str(string, None)
polar/ffi.py:26: in load_str
    check_result(lib.polar_load(self.ptr, string, filename))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

result = 0

    def check_result(result):
        if result == 0 or is_null(result):
>           raise Error().get()
E           polar.exceptions.UnrecognizedToken: did not expect to find the token ':' at line 1, column 23

polar/ffi.py:108: UnrecognizedToken
```